segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -943,13 +1246,29 @@ public boolean isStorageSystemSegment(String segmentName) {
     }
 
     private String getSystemJournalChunkName() {
-        return getSystemJournalChunkName(containerId, epoch, currentFileIndex);
+        return getSystemJournalChunkName(containerId, epoch, currentFileIndex.get());
     }
 
     private String getSystemJournalChunkName(int containerId, long epoch, long currentFileIndex) {
         return NameUtils.getSystemJournalFileName(containerId, epoch, currentFileIndex);
     }
 
+    private <R> CompletableFuture<R> executeSerialized(Callable<CompletableFuture<R>> operation) {
+        return this.taskProcessor.add(Collections.singletonList(LOCK_KEY_NAME), () -> executeExclusive(operation));
+    }
+
+    private <R> CompletableFuture<R> executeExclusive(Callable<CompletableFuture<R>> operation) {
+        return CompletableFuture.completedFuture(null).thenComposeAsync(v -> {
```

Comment:
Why do you have `CompletableFuture.completedFuture(null).thenComposeAsync`?

Is it because you want your `Callable` to execute async? Why can't it execute sync? It still returns a Future; it doesn't mean that upstream code needs to block on it.

Reply:
There is no CompletableFuture.composeAsync